Console errors

yakov116 opened the pull request:
@jerone r u in the mood of sending this?

You can take the credit (I hate console errors 😄 )

<!--

Thanks for contributing! 🍄

1. LINKED ISSUES:
   Does this PR close/fix an existing issue? Write something like `Closes #10`

2. TEST URLS:
   Add some test URLs

3. SCREENSHOT:
   Add a screenshot here if your PR makes visual changes

yakov116 committed:
Make `repo-age` work the first time you visit a repo

To test the timeout reason https://github.com/yakov116/blank

jerone commented:
Sorry, I lost interest in that project, as it was really hard to contribute.

yakov116 commented:
Ok thanks any ways

yakov116 commented:
@FloEdelmann The maintainer felt that my PR's we not up to par, so I am not sending pr's anymore.

Would you want to send it? (I tested everything, but make sure to test again)

FloEdelmann commented:
@yakov116 I can't reproduce the errors for `clean-issue-filter` and `clear-pr-merge-commit-message` / `fit-textareas` / `suggest-commit-title-limit` using the current version of Refined GitHub. Do I have to click/do something to trigger the errors?

Regarding the `repo-age` fix: What is it supposed to fix?

I can submit the `hide-disabled-milestone-sorter` fix, though.

yakov116 commented:
>I can't reproduce the errors for clean-issue-filter 
Test on  fregante/slash-tag-test/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc
```js
Uncaught (in promise) TypeError: Cannot read property 'parentElement' of undefined
    at hideMilestones (refined-github.js:11936)
Uncaught (in promise) TypeError: Cannot read property 'parentElement' of undefined
    at hideProjects (refined-github.js:11954)
```
'#js-issues-toolbar' does not exist when there are no issues.

>Regarding the repo-age fix: What is it supposed to fix?
Visit any repo for the first time. ex: https://github.com/fregante/slash-tag-test

```js
❌ Refined GitHub → repo-age → TypeError: Cannot read property 'src' of undefined
    at refined-github.js:13475
    at async getSet (refined-github.js:2265) 
Search issue: https://github.com/sindresorhus/refined-github/issues?q=is%3Aissue+Cannot%20read%20property%20'src'%20of%20undefined

Open an issue: https://github.com/sindresorhus/refined-github/issues/new?labels=bug&template=bug_report.md&title=%60repo-age%60%3A%20Cannot%20read%20property%20'src'%20of%20undefined
```
It fires too early. See https://github.com/sindresorhus/refined-github/issues/2907#issuecomment-620556750

Rearding the time out. On a new repo when there si no commits, it will never stop since the repo bar does not exist.
To test the timeout reason yakov116/blank

>I can't reproduce the errors for clean-issue-filter and clear-pr-merge-commit-message / fit-textareas / suggest-commit-title-limit

Hmm your right it stopped happening

FloEdelmann commented:
I created PRs for `clean-issue-filters` and `hide-disabled-milestone-sorter`: https://github.com/sindresorhus/refined-github/pulls

I still can't reproduce `repo-age` though: When I go to https://github.com/fregante/slash-tag-test (also after cleaning the extension cache), no error is thrown. I tried it in both Firefox and Chromium.

yakov116 commented:
> I created PRs for `clean-issue-filters` and `hide-disabled-milestone-sorter`: [sindresorhus/refined-github/pulls](https://github.com/sindresorhus/refined-github/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc)

Thanks!


> I still can't reproduce `repo-age` though: When I go to [fregante/slash-tag-test](https://github.com/fregante/slash-tag-test) (also after cleaning the extension cache), no error is thrown. I tried it in both Firefox and Chromium.

Weird I have it always.

If you want there is also https://github.com/yakov116/refined-github/pull/5

yakov116 commented on source/features/hide-disabled-milestone-sorter.tsx:
```diff
@@ -21,5 +17,10 @@ features.add({
 	],
-	init
+	exclude: [
+		() => !select.exists('.js-milestone-issues-container')
```
@FloEdelmann curious why you didnt do it like this?

You eliminate a whole function.

Just asking.

Side note: Thanks

FloEdelmann replied:
We need to select the element anyway to check if it exists. If it does, we can directly pass the element instead of its selector to `observeElement`, so it doesn't have to do the `select` again.

yakov116 replied:
Right. But my question was https://github.com/yakov116/refined-github/pull/4/commits/b1f16963da253f6d91fc08cb01a5d431c463ee7d

By doing the `observeElement` in the init you eliminate the entire extra function.

FloEdelmann replied:
It's still a function, just an anonymous arrow function instead :D

I don't really care where the `select` and `observeElement` calls are, but why should I change it if the arrow function style has no real advantages?

yakov116 replied:
I am talking about the hide function.

But you answer still applies
> why should I change it if the arrow function style has no real advantages?

yakov116 replied:
Thanks!!

Are you willing to do #5?

Side Note: You made me very happy by sending the PR's

FloEdelmann replied:
I don't know what #5 is supposed to do. But I see that the limit increased from 1 to 100, which doesn't seem like the right way to do the query.

yakov116 replied:
https://github.com/sindresorhus/refined-github/issues/2934#issuecomment-604146813

yakov116 replied:
You should be able to cherry pick that. Double check the types

yakov116 replied:
You should be able to cherry pick that. Double check the types

FloEdelmann replied:
I feel unconfident to submit a PR for this one, because I can't easily reproduce it, and don't really know how the feature is working.

yakov116 replied:
Do you have any org. If yes fork refined-github to your org and you will see "2 open pull requests"

yakov116 committed:
Only run `show-open-prs-of-forks` on settings page

Prevents this.
❌ Refined GitHub → show-open-prs-of-forks → TypeError: Cannot read property 'after' of null
    at initDeleteHint (refined-github.js:13940)
    at async runFeature (refined-github.js:5445)
    at async setupPageLoad (refined-github.js:5459) 
Search issue: https://github.com/sindresorhus/refined-github/issues?q=is%3Aissue+Cannot%20read%20property%20'after'%20of%20null

yakov116 commented:
@FloEdelmann I added one more 
Only run `show-open-prs-of-forks` on settings page
Go to your refined github fork and go to settings page others than the options page
